### DISCUSSION TITLE
Using a secure socket with uasyncio

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#10179] [Libraries & Drivers] [answered]
Hello,
Does anyone know if it is possible to setup a TLS server with uasyncio?
I use `await asyncio.start_server(cb, host, port)` and it works fine, but it seems there is no support for a secure socket. If I am not mistaking Micropython has already support to wrap a socket to use SSL. (using function `ssl.wrap_socket`).

**Comment by jimmo** [comment, marked as answer]
This is not yet implemented. Many of the required dependencies are done, but see #8968 for some recent work on adding SSLContext support (to replace the non-compatible wrap_socket implementation in ussl).
